### PR TITLE
grafana/ui: Add default type="button" to <Button>

### LINKED ISSUE #41863: grafana/ui: Add default type="button" to <Button>
Currently our Button component in grafana/ui does not specify a type attribute. By the specification this defaults to `submit` https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button#attr-type. This creates quite a number of bugs, because anytime somebody adds a button in data source config page, which is wrapped in `<form>` (and probably other contexts) that button will act as submit button for the form, see for example [this pr](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button#attr-type).

As the submit case seems to be minority usecase and this generates lots of bugs, I think it would make sense to default to `type="button"` instead when not set. This would be a breaking change but I think it should not affect that much and relying on the default behavior should be easily checked at least in our code base (basically only where we use `<form>`)

### PR DESCRIPTION
**Pull request by axelavargas**
**What this PR does / why we need it**:

Change the default to type="button"  in the `Button` component.

**Which issue(s) this PR fixes**:

Fixes #41863

**Special notes for your reviewer**:

# Release notice breaking change

**grafana/ui: Button now specifies a default type="button"**

The `Button` component provided by @grafana/ui now specifies a default `type="button"` when no type is provided. In previous versions, if the attribute was not specified for buttons associated with a `<form>` the default value was `submit` per the [specification](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button#attr-type)

You can preserve the old behavior by explicitly setting the type attribute: `<Button type="submit" />`

[Github Issue #41863](https://github.com/grafana/grafana/issues/41863).

### CODE REVIEW

**Comment by axelavargas** [comment]
> 

Yes, I need to do that as well, I just saw unit tests failing because of that 🙈

File: public/app/features/alerting/unified/components/receivers/ReceiversSection.tsx [hunk 1]
```diff
@@ -33,7 +33,9 @@ export const ReceiversSection: FC<Props> = ({
         </div>
         {showButton && (
           <Link to={addButtonTo}>
-            <Button icon="plus">{addButtonLabel}</Button>
+            <Button type="submit" icon="plus">
```
Comment (gillesdemey) [inline public/app/features/alerting/unified/components/receivers/ReceiversSection.tsx:36]: This one should be a `type="button"`, I see it's also nested in a `<a>` tag, which is illegal :/

Comment (axelavargas) [reply public/app/features/alerting/unified/components/receivers/ReceiversSection.tsx:36]: good catch @gillesdemey 🙌🏾, I will double-check again these changes

**Review by gillesdemey — APPROVED** [review]
The changes in `features/alerting/unified` look good! 🚢

**Comment by tolzhabayev** [comment]
@axelavargas definitely add a changelog label to this one and add it to the [announcements](https://community.grafana.com/c/plugin-development/plugin-announcements/47), there are `42` plugins [using](https://ops.grafana.net/d/dmb2o0xnz/imported-property-details?orgId=1&var-propertyName=Button&var-packageName=@grafana/ui) this component.

**Comment by axelavargas** [comment]
> @axelavargas definitely add a changelog label to this one and add it to the [announcements](https://community.grafana.com/c/plugin-development/plugin-announcements/47), there are `42` plugins [using](https://ops.grafana.net/d/dmb2o0xnz/imported-property-details?orgId=1&var-propertyName=Button&var-packageName=@grafana/ui) this component.

Thanks, @tolzhabayev! , good catch 😄

Also, I was thinking about the idea that @aocenas mentioned about automating and creating an issue in the plugin repositories that are using this component,  saying something like "you use grafana/button we are doing this change `X` in Grafana V9.0, make sure your plugin still works".  I am not sure if this is something we did in the past 🤔.  ps, I don't have yet an approach to how we could automate this, and when we should do it, but If we think is a good idea I can spend some time next week after this PR is merged :).

**Comment by tolzhabayev** [comment]
> > @axelavargas definitely add a changelog label to this one and add it to the [announcements](https://community.grafana.com/c/plugin-development/plugin-announcements/47), there are `42` plugins [using](https://ops.grafana.net/d/dmb2o0xnz/imported-property-details?orgId=1&var-propertyName=Button&var-packageName=@grafana/ui) this component.
> 
> Thanks, @tolzhabayev! , good catch 😄
> 
> Also, I was thinking about the idea that @aocenas mentioned about automating and creating an issue in the plugin repositories that are using this component, saying something like "you use grafana/button we are doing this change `X` in Grafana V9.0, make sure your plugin still works". I am not sure if this is something we did in the past 🤔. ps, I don't have yet an approach to how we could automate this, and when we should do it, but If we think is a good idea I can spend some time next week after this PR is merged :).

it was not done before to my knowledge, but if we can script that - it might be worth it. Basically it is 
- Getting the list from our data / bigQuery of all the plugin ID's that are affected
- Using the GCOM [API](https://grafana.com/api/plugins) getting the github url of the plugin (it's in the `url` field 
- Using github api or cli to open an issue

**Review by JoaoSilvaGrafana — APPROVED** [review]
Looks great! Did some testing and from what I can tell it works well, even fixes an issue where for example on a Prometheus data source configuration if you pressed enter it would add or remove an exemplar, now enter correctly saves the data source 👏

**Review by joshhunt — APPROVED** [review]
Took this for a brief little whirl configuring a dashboard, and using the Azure Data Explorer query editor, and found no issues. Well done 👍🏻

**Pull request merged**